agent/stats/engine.go

```diff
@@ -433,15 +431,20 @@ func (engine *DockerStatsEngine) GetInstanceMetrics(includeServiceConnectStats b
 	engine.lock.Lock()
 	defer engine.lock.Unlock()
 
+	if includeServiceConnectStats {
+		err := engine.getServiceConnectStats()
+		if err != nil {
+			seelog.Errorf("Error getting service connect metrics: %v", err)
```

Comment:
hmm, is this the best we can do with this error?

Reply:
We could have retries but I feel `getServiceConnectStats()` is not the right place for retries on error as it pulls service connect metrics for all the tasks in parallel. This could be a better place for the retry logic - https://github.com/aws/amazon-ecs-agent/blob/feature/service-connect/agent/stats/service_connect_linux.go#L46

What are your thoughts? @angelcar

Reply:
sgtm

Reply:
Added a TODO to keep track.